src/views/Home/hooks/useGetTopPoolsByApr.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect, useMemo } from 'react'
+import { usePriceCakeBusd } from 'state/farms/hooks'
+import { useAppDispatch } from 'state'
+import BigNumber from 'bignumber.js'
+import { orderBy } from 'lodash'
+import { Pool, State } from 'state/types'
+import { fetchCakeVaultFees, fetchPoolsPublicDataAsync } from 'state/pools'
+import { simpleRpcProvider } from 'utils/providers'
+import { useSelector } from 'react-redux'
+import { useCakeVault } from 'state/pools/hooks'
+import { getAprData } from 'views/Pools/helpers'
+
+enum FetchStatus {
+  NOT_FETCHED = 'not-fetched',
+  FETCHING = 'fetching',
+  SUCCESS = 'success',
+  FAILED = 'failed',
+}
+
+const useGetTopPoolsByApr = (isIntersecting: boolean) => {
+  const dispatch = useAppDispatch()
+  const { pools: poolsWithoutAutoVault } = useSelector((state: State) => ({
+    pools: state.pools.data,
+    userDataLoaded: state.pools.userDataLoaded,
+  }))
+  const {
+    fees: { performanceFee },
+  } = useCakeVault()
+  const performanceFeeAsDecimal = performanceFee && performanceFee / 100
+  const [fetchStatus, setFetchStatus] = useState(FetchStatus.NOT_FETCHED)
+  const [topPools, setTopPools] = useState<Pool[]>([null, null, null, null, null])
```

Comment:
Why the `null` placeholders?

Reply:
So that I can hold all of the 'loading' state management within the `TopFarmPool` component (eugh that name.. shudder), while still mapping over that array to render the components.

The alternative as I saw it would be to declare a null state and render a separate 'loading' component with 5 different loading skeletons in it until that null state was populated. 

I think the way I have it requires less code, but is slightly less intuitive?